server/src/main/scala/controllers/application/WithImports.scala

```diff
@@ -31,17 +33,31 @@ trait WithImports {
       log.info(s"Received a request to import red list counts")
       request.body.asJson match {
         case Some(content) =>
-          log.info(s"Received red list pax data")
+          log.info(s"Received red list pax data $content")
           Try(content.toString.parseJson.convertTo[RedListCounts])
-            .map(redListCounts => ctrl.flightsActor
-              .ask(redListCounts)
-              .map(_ => Accepted(toJson(ApiResponseBody(s"${redListCounts.counts.size} red list records imported"))))
-            ).getOrElse(Future.successful(BadRequest("Failed to parse json")))
+            .map { redListCounts =>
+              getRedListCount(redListCounts)
+                .onComplete {
+                  case Success(_) => ctrl.flightsActor
+                    .ask(redListCounts)
+                  case Failure(exception) =>
+                    log.error(s"Error $exception ")
+                }
+              Future.successful(Accepted(toJson(ApiResponseBody(s"${redListCounts.passengers.size} red list records imported"))))
+            }.getOrElse(Future.successful(BadRequest("Failed to parse json")))
         case None => Future.successful(BadRequest("No content"))
       }
     }
   }
 
+  def getRedListCount(redListCounts: RedListCounts): Future[Iterable[Any]] = Future.sequence {
```

Comment:
A couple of improvements we can make here:

1. The function doesn't 'get' the red list counts, but rather sends them to the red list pax actors.
2. If you change the actor's behaviour so that it responds with its updated state then we can send this state to the arrivals actor
3. We can then remove the code in the arrivals actor to look up the red list counts